src/sentry/api/endpoints/organization_member_team_details.py

```diff
@@ -150,6 +187,43 @@ def post(
 
         return Response(serialize(team, request.user, TeamWithProjectsSerializer()), status=201)
 
+    def put(
+        self,
+        request: Request,
+        organization: Organization,
+        member: OrganizationMember,
+        team_slug: str,
+    ) -> Response:
+        try:
+            team = Team.objects.get(organization=organization, slug=team_slug)
+        except Team.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        try:
+            omt = OrganizationMemberTeam.objects.get(team=team, organizationmember=member)
+        except OrganizationMemberTeam.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        serializer = OrganizationMemberTeamSerializer(data=request.data, partial=True)
+        if not serializer.is_valid():
+            return Response(status=400)
+        result = serializer.validated_data
+
+        if "role" in result:
+            if features.has("organizations:team-roles", organization):
+                new_role_id = result["role"]
+                try:
+                    new_role = team_roles.get(new_role_id)
+                except KeyError:
+                    return Response(status=400)
+
+                if not self._can_set_team_role(request, team, new_role):
+                    return Response({"detail": ERR_INSUFFICIENT_ROLE}, status=400)
+
+                omt.update_team_role(new_role)
+
+        return Response(serialize(team, request.user, TeamWithProjectsSerializer()), status=200)
```

Comment:
This (`TeamWithProjectsSerializer`) seems like the wrong thing to display at this level of the API. I put it in for the sake of consistency with the POST method. But it's describing the entire team (redundantly to another endpoint?) rather than just the membership object.

Thoughts on creating a new serializer for `OrganizationMemberTeamDetailsEndpoint` and on whether we want to change existing behavior on POST?

Reply:
Having the resource be consistent with other methods feels like the right thing to do to me. This `PUT` method could operate on other attributes in the future and reflecting back the entity that was changed is likely what we'd want.

Do we need the `projects` in this response? I thought we were trying to move away from automatically including 1:N relations.

Reply:
> Having the resource be consistent with other methods feels like the right thing to do to me. This `PUT` method could operate on other attributes in the future and reflecting back the entity that was changed is likely what we'd want.

👍

> Do we need the `projects` in this response? I thought we were trying to move away from automatically including 1:N relations.

I'll look into whether it's safe to take `projects` out of the POST response also.